cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaMirrorMaker2Cluster.java

```diff
@@ -142,9 +146,16 @@ private static KafkaConnectSpec buildKafkaConnectSpec(KafkaMirrorMaker2Spec spec
                 .withMetricsConfig(spec.getMetricsConfig())
                 .withTracing(spec.getTracing())
                 .withTemplate(spec.getTemplate())
-                .withExternalConfiguration(spec.getExternalConfiguration())
-                .build();
+                .withExternalConfiguration(spec.getExternalConfiguration());
+
+        Rack rack = spec.getRack();
+        if (rack != null) {
+            kafkaConnectSpecBuilder
+                    .withRack(rack)
+                    .withImage(System.getenv().getOrDefault(STRIMZI_DEFAULT_KAFKA_INIT_IMAGE, DEFAULT_STRIMZI_OPERATOR_IMAGE));
+        }
 
+        return kafkaConnectSpecBuilder.build();
```

Comment:
Is this needed? Why can't you just do `withRack(spec.getRack)` and the same for the image? It would be either `null` or have the right value